atd-etl/cris_import/cris_import.py

```diff
@@ -276,15 +392,48 @@ def unzip_archives(archives_directory):
     containing an archive's contents
     """
 
-    extracted_csv_directories = []
-    for filename in os.listdir(archives_directory):
-        print("About to unzip: " + filename + "with the command ...")
-        extract_tmpdir = tempfile.mkdtemp()
-        unzip_command = f'7za -y -p{ZIP_PASSWORD} -o"{extract_tmpdir}" x "{archives_directory}/{filename}"'
-        print(unzip_command)
-        os.system(unzip_command)
-        extracted_csv_directories.append(extract_tmpdir)
-    return extracted_csv_directories
+    with SshKeyTempDir() as key_directory:
+        write_key_to_file(
+            key_directory + "/id_ed25519", DB_BASTION_HOST_SSH_PRIVATE_KEY + "\n"
+        )
+        ssh_tunnel = SSHTunnelForwarder(
+            (DB_BASTION_HOST),
+            ssh_username=DB_BASTION_HOST_SSH_USERNAME,
+            ssh_private_key=f"{key_directory}/id_ed25519",
+            remote_bind_address=(DB_RDS_HOST, 5432),
+        )
+        ssh_tunnel.start()
+
+        pg = psycopg2.connect(
+            host="localhost",
+            port=ssh_tunnel.local_bind_port,
+            user=DB_USER,
+            password=DB_PASS,
+            dbname=DB_NAME,
+            sslmode=DB_SSL_REQUIREMENT,
+            sslrootcert="/root/rds-combined-ca-bundle.pem",
+        )
+
+        extracted_csv_directories = []
+        for filename in os.listdir(archives_directory):
+            print("About to unzip: " + filename + "with the command ...")
+            extract_tmpdir = tempfile.mkdtemp()
+            unzip_command = f'7za -y -p{ZIP_PASSWORD} -o"{extract_tmpdir}" x "{archives_directory}/{filename}"'
+            print(unzip_command)
+            os.system(unzip_command)
+
+            cursor = pg.cursor()
+            cursor.execute(
+                "select id from cris_import_log where object_name = %s order by created_at desc limit 1",
+                (filename,),
+            )
+            id = cursor.fetchone()[0]
+            extracted_csv_directories.append((id, extract_tmpdir, filename))
+
+            pg.commit()
+            pg.close()
```

Comment:
while testing with multiple extract in the directory, this block threw an error because `pg.close()` closes the connection after the first file is processed. the fix is to de-indent `pg.close()` 👍

Reply:
Great catch, thank you. I've outdented those back into their correct block. https://github.com/cityofaustin/atd-vz-data/pull/1445/commits/b9ce5eb4f5350148340d035d2bbf510654526e58